bazel/p4_library.bzl

```diff
@@ -2,26 +2,62 @@
 
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
 
-def _p4_library_impl(ctx):
-    p4c = ctx.executable._p4c
+def _extract_common_p4c_args(ctx):
```

Comment:
Could we make this function only do one thing, i.e. extract the common p4c args, as the name suggests?
I think that would make things easier to understand, and extracting the inputs is just a one-liner that we can either duplicate or make a separate function.

Reply:
I separated extracting the inputs to a helper function `_extract_p4c_inputs` that is called by the build rule implementations